### PR TITLE
Better handling of addLabel

### PR DESCRIPTION
**Pull request by anandsinghparihar**
https://particl.atlassian.net/browse/PAR-245

### CODE REVIEW

File: src/app/wallet/receive/modals/add-address-label/add-address-label.component.ts [hunk 1]
```diff
@@ -43,7 +43,11 @@ export class AddAddressLabelComponent implements OnInit {
   onSubmitForm(): void {
     if (this.rpc.state.get('locked')) {
       // unlock wallet
-      this._modals.open('unlock', {forceOpen: true, timeout: 3, callback: this.addNewLabel.bind(this)});
+      if (this.type === 'private') {
+        this._modals.open('unlock', {forceOpen: true, timeout: 3, callback: this.addNewLabel.bind(this)});
+      } else {
```
Comment (kewde) [inline src/app/wallet/receive/modals/add-address-label/add-address-label.component.ts]: Adding public addresses also requires unlock

Comment (anandsinghparihar) [reply src/app/wallet/receive/modals/add-address-label/add-address-label.component.ts]: I guess no ...

Comment (kewde) [reply src/app/wallet/receive/modals/add-address-label/add-address-label.component.ts]: I got this confused with the address book manager, seems like getting new private addresses does require an unlock.

**Comment by anandsinghparihar** [comment]
@kewde please dont merge it I've some changes let me add in this branch

File: src/app/modals/modals.component.ts [hunk 2]
```diff
@@ -74,7 +74,7 @@ export class ModalsComponent implements DoCheck, OnInit {
     this.modal = this.modalContainer.createComponent(factory);
     const dynamicModal = this.modal as any;
     if (data !== undefined && dynamicModal.instance.setData !== undefined) {
-      dynamicModal.instance.setData(data);
+      dynamicModal.instance.setData(data.data);
```
Comment (kewde) [inline src/app/modals/modals.component.ts]: Please remove this,
just remove the {data: data} -> data in modal.service

Comment (anandsinghparihar) [reply src/app/modals/modals.component.ts]: Okay let me replace it  :)

**Pull request merged**